src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ComCaseloadService.kt

```diff
@@ -108,11 +120,19 @@ class ComCaseloadService(
 
   private fun filterOffendersEligibleForLicence(cases: List<ManagedCase>): List<ManagedCase> {
     val eligibleOffenders = cases.filter { case ->
-      case.nomisRecord?.prisonerNumber != null &&
-        comCaseloadSearchService.getIneligibilityReasons(case.nomisRecord)
-          .isEmpty()
+      val prisoner = case.nomisRecord?.toPrisonerSearchPrisoner()
+      when {
+        prisoner == null || prisoner.bookingId == null -> false
+        !eligibilityService.isEligibleForCvl(prisoner) -> false
+        else -> true
+      }
+    }
+    val hdcStatuses = hdcService.getHdcStatus(eligibleOffenders.map { it.nomisRecord!!.toPrisonerSearchPrisoner() })
+    return eligibleOffenders.filter {
+      val kind = it.findRelevantLicence()?.kind
+      val bookingId = it.nomisRecord?.bookingId?.toLong()!!
+      hdcStatuses.canBeSeenByCom(kind, bookingId)
```

Comment:
This is the main purpose of the ticket:
* don't do a call to nomis for every hdc checks
* take into account the kind - so we can return approved HDC cases in the COM create caseload